Amplify/Categories/DataStore/Model/Collection/List+Model.swift

```diff
@@ -85,35 +92,35 @@ open class List<ModelType: Model>: ModelList {
             }
         }
 
-        let json = try JSONValue(from: decoder)
-
-        switch json {
-        case .array:
-            let elements = try Elements(from: decoder)
-            self.init(elements)
-        default:
-            self.init(Elements())
-        }
+        self.init()
     }
 
-    public func encode(to encoder: Encoder) throws {
+    open func encode(to encoder: Encoder) throws {
+        let elements = [ModelType]()
         try elements.encode(to: encoder)
     }
 
     // MARK: - Asynchronous API
 
-    /// Use to initialize the collection.
+    /// Call this to initialize the collection if you have retrieved the list by traversing from your model objects
+    /// to its associated children objects. For example, a Post model may contain a list of Comments. By retrieving the
+    /// post object and trvasering to the comments, the comments are not retrieved from the data source until this
+    /// method is called. Data will be retrieved based on the plugin's data source and may have different failure
+    /// conditions, such as one that requires connectivity may fail with a network error. You can alternatively access
+    /// the Collection methods on the list to trigger the retrieval of data, but also expect that any operation to
+    /// perform as fast as it takes to make a request to the datastore and back.
```

Comment:
Suggested reword. Please format comment appropriately.

```suggestion
    /// conditions--for example, a data source that requires network connectivity may fail if the network is unavailable.
    /// Alternately, you can trigger an implicit `load` by invoking
    /// the Collection methods (such as using `map`, or iterating in a `for/in` loop)  on the List, which will retrieve data if it hasn't already been retrieved. In such cases, the time to perform that operation will include the time required to request data from the underlying data source.
```